app/src/main/java/org/simple/clinic/teleconsultlog/teleconsultrecord/TeleconsultRecordRepository.kt

```diff
@@ -0,0 +1,49 @@
+package org.simple.clinic.teleconsultlog.teleconsultrecord
+
+import org.simple.clinic.storage.Timestamps
+import org.simple.clinic.util.UtcClock
+import org.simple.clinic.uuid.UuidGenerator
+import java.time.Instant
+import java.util.UUID
+import javax.inject.Inject
+
+class TeleconsultRecordRepository @Inject constructor(
+    private val uuidGenerator: UuidGenerator,
+    private val teleconsultRecordDao: TeleconsultRecord.RoomDao,
+    private val teleconsultRecordWithPrescribedDrugsDao: TeleconsultRecordWithPrescribedDrugs.RoomDao,
+    private val utcClock: UtcClock
+) {
+
+  fun getTeleconsultRecordWithPrescribedDrugs(teleconsultRecordId: UUID): TeleconsultRecordWithPrescribedDrugs? {
+    return teleconsultRecordWithPrescribedDrugsDao.getPrescribedUuidForTeleconsultRecordUuid(teleconsultRecordId)
+  }
+
+  fun createTeleconsultRecord(
+      teleconsultRecordId: UUID = uuidGenerator.v4(),
+      teleconsultationType: TeleconsultationType,
+      patientConsented: Answer,
+      patientTookMedicine: Answer,
+      patientUuid: UUID,
+      medicalOfficerId: UUID,
+      medicalOfficerRegistrationId: String?
```

Comment:
Most of the parameters are used to generate the `TeleconsultRecordInfo`, instead of creating another type I can move the construction on `TeleconsultRecordInfo` effect handler and pass that in this function. WDYT?

Reply:
I feel like adding more types to this feature can get confusing to maintain IMO

Reply:
That's a good point. Let's do what you suggested and move it to the effect handler.